# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/VideoUtils.cpp
# Commit: c92507498323
# Full Hash: c92507498323b9b75e1f45e44a06bf5fa0124d6a
# Author: Paul Adenot <paul@paul.cx>
# Date: 2025-05-14 21:17:00
# Description:
#   Bug 1965980 - Increase stack on Windows for libaom encoding again. r=media-playback-reviewers,alwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D249082
# ==============================================================================

diff -r 5211e49f26cb -r c92507498323 dom/media/VideoUtils.cpp
--- a/dom/media/VideoUtils.cpp	Wed May 14 13:38:52 2025 +0300
+++ b/dom/media/VideoUtils.cpp	Wed May 14 09:53:13 2025 +0000
@@ -274,13 +274,14 @@
       SharedThreadPool::Get(nsDependentCString(name), threads);
 
   // Ensure a larger stack for platform decoder threads
-  if (aType == MediaThreadType::PLATFORM_DECODER) {
-    uint32_t minStackSize = 512 * 1024;
-    #ifdef XP_WIN
-    // libaom can require a larger stack size on Windows, because it uses
-    // large stack buffers
-    minStackSize *= 4;
-    #endif
+  bool needsLargerStacks = aType == MediaThreadType::PLATFORM_DECODER;
+// On Windows, platform encoder threads require larger stacks as well for
+// libaom.
+#ifdef XP_WIN
+  needsLargerStacks |= aType == MediaThreadType::PLATFORM_ENCODER;
+#endif
+  if (needsLargerStacks) {
+    const uint32_t minStackSize = 512 * 1024;
     uint32_t stackSize;
     MOZ_ALWAYS_SUCCEEDS(pool->GetThreadStackSize(&stackSize));
     if (stackSize < minStackSize) {
